x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -275,9 +274,6 @@ public PlanFactory visitRenameCommand(EsqlBaseParser.RenameCommandContext ctx) {
 
     @Override
     public PlanFactory visitKeepCommand(EsqlBaseParser.KeepCommandContext ctx) {
-        if (ctx.PROJECT() != null) {
-            addWarning("PROJECT command is no longer supported, please use KEEP instead");
-        }
```

Comment:
We need to see if there are any blockers on removing this section, in the context of BWC. This being a warning (vs. error) tells me we need to check the aspect of bwc.

Reply:
This was checked in Kibana, but the team is already aware of this change.
Apart from that, we should not have other dependencies.
About strict BWC in terms of inter-node communication, we are be safe since this change only touches the grammar; the plan and the serialized data structures do not require any chanages.